### ISSUE TITLE
copydoc fails to copy the correct function overload for Doxygen 1.9.5 - trunk (7e1860534d163a9d8882ce3ebb620719f74e75eb)

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#11127]
**Describe the bug**
`copydoc` fails to copy the correct overload for Doxygen 1.9.5 - trunk (7e1860534d163a9d8882ce3ebb620719f74e75eb)

```
/**
 * X docs
 */
struct X
{
  /**
   * operator==(struct X&, struct X&)
   */
  friend bool operator==(struct X&, struct X&) = default;

  /**
   * operator==(struct X&, char)
   */
  friend bool operator==(struct X&, char);

  /**
   * \copydoc operator==(struct X&, struct X&)
   */
  friend bool operator==(struct X&, int);

  /**
   * \copydoc operator==(struct X&, char)
   */
  friend bool operator==(struct X&, long);
};
```

`\copydoc operator==(struct X&, char)` does not copy the documentation from the `operator==(struct X&, char)` overload. This bug applies to any operator including three-way comparison as well.

**Screenshots**
Doxygen 1.9.4
![image](https://github.com/user-attachments/assets/3b5d31ee-da6c-48aa-9c1e-39e20d5d40d5)

Doxygen 1.9.5 - trunk (7e1860534d163a9d8882ce3ebb620719f74e75eb)
![image](https://github.com/user-attachments/assets/d44a0970-9b7b-4102-ae2f-b8fd35434bbf)

**To Reproduce**
[testcase.zip](https://github.com/user-attachments/files/16937807/testcase.zip)

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by scott-zhong** [comment]
@albert-github @doxygen I have verified the fix. Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
